source/memory.tex

```diff
@@ -1379,9 +1379,8 @@
     template<class T> using rebind_alloc = @\seebelow@;
     template<class T> using rebind_traits = allocator_traits<rebind_alloc<T>>;
 
-    [[nodiscard]] static constexpr pointer allocate(Alloc& a, size_type n);
-    [[nodiscard]] static constexpr pointer allocate(Alloc& a, size_type n,
-                                                    const_void_pointer hint);
+    static constexpr pointer allocate(Alloc& a, size_type n);
+    static constexpr pointer allocate(Alloc& a, size_type n, const_void_pointer hint);
     [[nodiscard]] static constexpr allocation_result<pointer, size_type>
```

Comment:
Obviously, this line (and line 1574) also needs to be changed.

> [allocator.traits.other] No such declaration for allocation_result - change not applied.

The standard wording referenced in [P2422R1](https://wg21.link/p2422r1) predated [P2625R2](https://wg21.link/p2652r2), so `allocate_at_least` was mentioned in the old form.

Reply:
Yes, the intent of the paper is to remove `[[nodiscard]]`, including on the `allocate_at_least` function. Just because it's moved from [allocator.traits.other] to [allocator.traits.members] doesn't mean the `[[nodiscard]]` should be left in place.

LWG should have spotted this though.

Reply:
Added as a separate commit - please check.